src/llmq/chainlocks.cpp

```diff
@@ -219,11 +219,9 @@ void CChainLocksHandler::UpdatedBlockTip()
 
 void CChainLocksHandler::CheckActiveState()
 {
-    const bool fDIP0008Active = WITH_LOCK(cs_main, return (m_chainstate.m_chain.Tip() != nullptr) && (m_chainstate.m_chain.Tip()->pprev != nullptr) && m_chainstate.m_chain.Tip()->pprev->nHeight >= Params().GetConsensus().DIP0008Height);
-
     bool oldIsEnforced = isEnforced;
     isEnabled = AreChainLocksEnabled(spork_manager);
-    isEnforced = (fDIP0008Active && isEnabled);
+    isEnforced = isEnabled.load();
```

Comment:
I don't think you can simply drop it like that - we should only enforce it once dip0008 is active, relying on spork alone is not enough.

Reply:
Why?

The only thing this function really does is clear out data in the case of spork going from off -> on. What is the value of gating that behind DIP8 activation?

Reply:
`isEnforced` is used in many places. The idea is that chainlocks could be enabled ahead of dip0008 activation but they would only be enforced once dip0008 is actually active to avoid network split between new and old nodes.

Reply:
This migration logic is no longer useful though... (and we have no test for it) but I don't like it being dropped only partially. I think we could refactor it i.e. `s/isEnforced/isEnabled/` and drop `isEnforced`.

Reply:
done